src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs

```
@@ -2447,10 +2447,28 @@ void M()
 }
 ";
             var comp = CreateCompilationWithMscorlib(source, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void EmbeddedDeclarationIsScoped()
+        {
+            var source = @"
+class C1
+{
+    void M()
+    {
+        if (true)
+            var (x, y) = (1, 2);
+        System.Console.WriteLine(x);
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, references: s_valueTupleRefs);
             comp.VerifyDiagnostics(
-                // (7,13): error CS1023: Embedded statement cannot be a declaration or labeled statement
-                //             var (x, y) = (1, 2);
-                Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "var (x, y) = (1, 2);").WithLocation(7, 13)
```

>Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "var (x, y) = (1, 2);").WithLocation(7, 13) [](start = 16, length = 85)

I think we might still want to report this error.